### PR TITLE
Groovy Language Cheat sheet

### PR DESCRIPTION
**Pull request by stanly-johnson**
**What does your Instant Answer do?**
A Cheat Sheet for Groovy-Language

**What is the data source for your Instant Answer? (Provide a link if possible)**
Groovy-Language,
sourceUrl: http://www.groovy-lang.org/syntax.html

**Why did you choose this data source?**
Reliable and easy-to-understand

**Are there any other alternative (better) data sources?**
No

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Programmers

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
https://duck.co/ideas/idea/6153/groovy-language-cheat-sheet

**Are you having any problems? Do you need our help with anything?**
Not yet

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
groovy2
groovy1

-----
IA Page: https://duck.co/ia/view/groovy_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/groovy.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
+    "id": "groovy_cheat_sheet",
+    "name": "Groovy",
+    "description": "Groovy cheat Sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/groovy.json]: Please remove this, it's a little redundant.

File: share/goodie/cheat_sheets/json/groovy.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    "metadata": {
+        "sourceName": "Groovy-Language",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/groovy.json]: Let's change this to "Groovy-lang" to match the domain

File: share/goodie/cheat_sheets/json/groovy.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      "val":"Uncapped floating point number"
+    }, {
+      "key":"java.util.List",
+      "val":"Contain values of any type"
+    }, {
+      "key":"java.util.Map",
+      "val":"No need for quotes"
+    }
+  ],
+  "Operators": [{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/groovy.json]: Whoops, please make sure you indentation is correct.

File: share/goodie/cheat_sheets/json/groovy.json [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    "key":"+",
+    "val":"a.plus(b)"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/groovy.json]: it wasn't clear if these were the Groovy operator or not. Let's use the words "Addition", "Subtraction", "Power", "Modulo", etc. to make it clear

**Comment by moollaza** [comment]
@stanly-johnson thanks a lot for contributing, we really appreciate it! I've left you some feedback that needs to be implemented. Once things are updated please let me so so I can review this again. If you have any questions, feel free to ask them here!

**Comment by stanly-johnson** [comment]
@moollaza

**Comment by moollaza** [comment]
@stanly-johnson congrats! It's live: https://duckduckgo.com/?q=groovy+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by stanly-johnson** [comment]
@moollaza Thank You :)

**Pull request merged**